### DISCUSSION TITLE
Ok to store a reference to `StateNotifier` in `StatefulWidget.state`?

### DISCUSSION THREAD

**Discussion by wujek-srujek** [rrousselGit/riverpod#572] [Q&A] [answered]
I have a `StatefulWidget`, and in `initState`, among other things, I `read` a `StateNotifier`:
```dart
_myNotifier = context.read(
  _myNotifierProvider.notifier,
);
```
and save it for later. I do this because in this page I need this notifier in many event handlers, and it got tedious to have to call `read` over and over again.

Is this something that is Ok and safe to do, or am I inadvertently breaking something without knowing it?

**Comment by rrousselGit** [comment, marked as answer]
This could get tricky if somehow the notifier can get rebuilt (like with context.refresh or if it uses ref.watch).

I would recommend keeping the boilerplate.

**Comment by otto-dev** [comment]
Just write a `getter` that calls `read` instead, so you don't have to repeat the boilerplate, but you also don't have to worry about stale references.

https://dart.dev/guides/language/language-tour
